lib/generators/data_migration_generator.rb

```diff
@@ -0,0 +1,16 @@
+class DataMigrationGenerator < Rails::Generators::NamedBase
+  source_root File.expand_path('templates', __dir__)
+
+  desc 'This generates a data migration service in app/services/data_mirations and adds it to the data migration rake task'
+  def create_data_migration_file
+    name = "DataMigrations::#{file_name.camelize}"
+    copy_file 'data_migration.rb.tt', "app/services/data_migrations/#{file_name}.rb"
+    gsub_file "app/services/data_migrations/#{file_name}.rb", '%{service_name}', file_name.camelize
+    gsub_file "app/services/data_migrations/#{file_name}.rb", '%{timestamp}', Time.zone.now.to_s(:number)
+
+    copy_file 'data_migration_spec.rb.tt', "spec/services/data_migrations/#{file_name}_spec.rb"
+    gsub_file "spec/services/data_migrations/#{file_name}_spec.rb", '%{service_name}', file_name.camelize
+
+    inject_into_file 'lib/tasks/data.rake', "  '#{name}',\n", after: "# do not delete or edit this line - services added below by generator\n"
```

Comment:
If I try to generate a similar service name and overwrite the old one, when I get to this stage I'm getting the following message:
```
File unchanged! The supplied flag value not found!  lib/tasks/data.rake
```
not sure if we can instead allow the message returned to be more user friendly and just indicate it hasn't changed because it already exists instead?

Reply:
This is what `thor` outputs, so I'm not sure if there is much I can do about it -  https://github.com/erikhuda/thor/issues/706 - There is also an open PR resolving this issue but it hasn't been merged yet.

You can replicate by running e.g. `rails g factory_bot:model DataMigration` twice.